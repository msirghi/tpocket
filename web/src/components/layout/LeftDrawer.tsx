import React from 'react';
import withLeftDrawer from "../../hocs/withLeftDrawer";
import Drawer from '@material-ui/core/Drawer';

interface IProps {
  open: boolean;
}

export const LeftDrawer: React.FC<IProps> = ({open}) => {
  return (
    <div>
      <Drawer
        // className={classes.drawer}
        variant="persistent"
        anchor="left"
        open={open}
        classes={{
          // paper: classes.drawerPaper,
        }}
      >
        salutare from drawer
      </Drawer>
    </div>
  );
}

// export default withLeftDrawer(LeftDrawer);
