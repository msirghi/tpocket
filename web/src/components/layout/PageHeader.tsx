import React from 'react';
import Typography from '@material-ui/core/Typography';

interface IProps {
  primaryText: string;
  secondaryText?: string;
}

export const PageHeader: React.FC<IProps> = ({ primaryText, secondaryText }) => {

  return (
    <div className={'text-center'} data-test={'header-wrapper'}>
      <Typography data-test={'primary-text'} variant="h5" component="h5" gutterBottom>
        { primaryText }
      </Typography>

      <Typography data-test={'secondary-text'} variant="subtitle1" gutterBottom>
        {secondaryText}
      </Typography>
    </div>
  );
}
