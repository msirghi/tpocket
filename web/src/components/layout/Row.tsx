import React from 'react';

interface IProps {
  className?: string;
}

export const Row: React.FC<IProps> = ({ className, children }) => {
  return (
    <div className={ `row ${ className }` }>
      { children }
    </div>
  );
}
