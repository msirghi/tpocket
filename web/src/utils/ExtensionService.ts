const monthNames: Array<string> = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December'
];

const getMonthName = (monthNumber: number) => monthNames[monthNumber];

export default {
  getMonthName
};
