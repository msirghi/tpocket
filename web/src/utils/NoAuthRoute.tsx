import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { getAccessToken } from '../accessToken';

// @ts-ignore
export const NoAuthRoute = ({ Component, ...rest }) => (
  <Route
    {...rest}
    render={(props) => (getAccessToken() ? <Redirect to='/home' /> : <Component {...props} />)}
  />
);
