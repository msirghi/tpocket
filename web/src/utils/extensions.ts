const passwordStrength = require('check-password-strength');

const monthNames: Array<string> = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];

export const getMonthName = (monthNumber: number) => monthNames[monthNumber];

export const validateEmailStr = (email: string) => {
  return !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(email);
};

export const isPasswordWeak = (password: string): boolean => {
  return passwordStrength(password).value === 'Weak';
};
