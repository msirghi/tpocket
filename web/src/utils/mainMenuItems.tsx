import React from "react";
import { IMenuItem } from "../interfaces/IMenuItem";
import HomeIcon from '@material-ui/icons/Home';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
import CategoryIcon from '@material-ui/icons/Category';
import HelpOutlineIcon from '@material-ui/icons/Help';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import EmailIcon from '@material-ui/icons/Email';

export const mainMenuItems: Array<IMenuItem> = [
  {
    id: 1,
    link: '/home',
    name: 'Home',
    icon: <HomeIcon className={ 'light-grey-icon' }/>
  },
  {
    id: 2,
    link: '/account',
    name: 'Account',
    icon: <AccountCircleIcon className={ 'light-grey-icon ' }/>
  },
  {
    id: 3,
    link: '/categories',
    name: 'Categories',
    icon: <CategoryIcon className={ 'light-grey-icon' }/>
  }
];

export const secondaryMenuItems: Array<IMenuItem> = [
  {
    id: 1,
    link: '/help',
    name: 'Help',
    icon: <HelpOutlineIcon className={ 'light-grey-icon' }/>
  },
  {
    id: 2,
    link: '/writeus',
    name: 'Support',
    icon: <EmailIcon className={ 'light-grey-icon' }/>
  },
  {
    id: 3,
    link: '/logout',
    name: 'Sign out',
    icon: <ExitToAppIcon className={ 'light-grey-icon ' }/>
  }
];
