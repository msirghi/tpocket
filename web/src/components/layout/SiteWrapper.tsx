import React, { useEffect, useState } from 'react';
import clsx from 'clsx';
import { createStyles, makeStyles, Theme, useTheme } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import List from '@material-ui/core/List';
import CssBaseline from '@material-ui/core/CssBaseline';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import MailIcon from '@material-ui/icons/Mail';
import Badge from "@material-ui/core/Badge";
import NotificationsIcon from "@material-ui/icons/NotificationsNone";
import AccountCircle from "@material-ui/icons/AccountCircle";
import { mainMenuItems, secondaryMenuItems } from "../../utils/mainMenuItems";
import { Link } from "react-router-dom";
import { useWindowSize } from "../../utils/useWindowSize";
import SwipeableDrawer from '@material-ui/core/SwipeableDrawer';

const drawerWidth = 240;

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: 'flex',
    },
    appBar: {
      zIndex: theme.zIndex.drawer + 1,
      transition: theme.transitions.create(['width', 'margin'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
      }),
    },
    appBarShift: {
      marginLeft: drawerWidth,
      width: `calc(100% - ${ drawerWidth }px)`,
      transition: theme.transitions.create(['width', 'margin'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.enteringScreen,
      }),
    },
    menuButton: {
      marginRight: 36,
    },
    hide: {
      display: 'none',
    },
    paper: {
      backgroundColor: '#233c46'
    },
    drawer: {
      width: drawerWidth,
      flexShrink: 0,
      whiteSpace: 'nowrap',
    },
    drawerOpen: {
      width: drawerWidth,
      transition: theme.transitions.create('width', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.enteringScreen,
      }),
    },
    drawerClose: {
      transition: theme.transitions.create('width', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
      }),
      overflowX: 'hidden',
      width: theme.spacing(7) + 1,
      [theme.breakpoints.up('sm')]: {
        width: theme.spacing(9) + 1,
      },
    },
    toolbar: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'flex-end',
      padding: theme.spacing(0, 1),
      ...theme.mixins.toolbar,
    },
    content: {
      width: '100%',
      flexGrow: 1,
      padding: theme.spacing(3),
      paddingLeft: window.innerWidth < 700 ? theme.spacing(0) : theme.spacing(3),
      paddingRight: window.innerWidth < 700 ? theme.spacing(0) : theme.spacing(3),
      overflowX: 'hidden'
    },
    title: {
      color: '#000'
    },
    sectionDesktop: {
      display: 'none',
      [theme.breakpoints.up('md')]: {
        display: 'flex',
      },
    },
    grow: {
      flexGrow: 1,
    },
  }),
);

const renderMenu = () => {
  return <>
    <List>
      { mainMenuItems.map(item => (
        <Link key={ item.id } to={ item.link }>
          <ListItem button key={ item.id }>
            <ListItemIcon key={ item.id }>
              { item.icon }
            </ListItemIcon>
            <ListItemText
              data-test={ `menu-${ item.name }` }
              key={ item.name } primary={ item.name }
              className={ 'drawer-text' }
            />
          </ListItem>
        </Link>
      )) }
    </List>
    <Divider/>
    <List>
      { secondaryMenuItems.map(item => (
        <Link key={ item.id } to={ item.link }>
          <ListItem button key={ item.id }>
            <ListItemIcon key={ item.id }>
              { item.icon }
            </ListItemIcon>
            <ListItemText key={ item.name } primary={ item.name } className={ 'drawer-text' }/>
          </ListItem>
        </Link>
      )) }
    </List>
  </>
}

export const SiteWrapper: React.FC = ({ children }) => {
  const classes = useStyles();
  const theme = useTheme();
  const [open, setOpen] = useState(false);
  const [mobileDrawerOpen, setMobileDrawerOpen] = useState(false);
  const [width, height] = useWindowSize();

  useEffect(() => {
    if (width < 900 && open) {
      console.log('entered')
      setOpen(false);
    }
  }, [width, height, open]);

  const handleDrawerOpen = () => {
    setOpen(true);
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };

  useEffect(() => {
    if (localStorage.getItem('drawerStatus') === 'null') {
      localStorage.setItem('drawerStatus', 'expanded');
      return;
    }

    setOpen(localStorage.getItem('drawerStatus') === 'expanded');
  }, []);

  useEffect(() => {
    localStorage.setItem('drawerStatus', open ? 'expanded' : 'hidden');
  }, [open]);

  const appBarStyle = width > 900 ? {
    [classes.appBarShift]: open,
  } : {};

  return (
    <div className={ classes.root } data-test={ 'site-wrapper' }>
      <CssBaseline/>
      <AppBar
        data-test={ 'site-header' }
        style={ { background: '#fff' } }
        position="fixed"
        className={ clsx(classes.appBar, appBarStyle) }
      >
        <Toolbar>
          <IconButton
            data-test={ 'menu-icon' }
            aria-label="open drawer"
            onClick={ () => width > 900 ? handleDrawerOpen() : setMobileDrawerOpen(true) }
            edge="start"
            className={ clsx(classes.menuButton, {
              [classes.hide]: open,
            }) }
          >
            <MenuIcon/>
          </IconButton>
          <Typography data-test={ 'site-name' } variant="h6" noWrap className={ classes.title }>
            TPocket
          </Typography>
          <div className={ classes.grow }/>
          <div className={ classes.sectionDesktop }>
            <IconButton aria-label="show 4 new mails">
              <Badge badgeContent={ 4 } color="primary">
                <MailIcon/>
              </Badge>
            </IconButton>
            <IconButton aria-label="show 17 new notifications">
              <Badge badgeContent={ 17 } color="primary">
                <NotificationsIcon/>
              </Badge>
            </IconButton>
            <IconButton
              edge="end"
              aria-label="account of current user"
              aria-haspopup="true"
            >
              <AccountCircle/>
            </IconButton>
          </div>
        </Toolbar>
      </AppBar>
      { width > 900 ? <Drawer
          variant="permanent"
          className={ clsx(classes.drawer, {
            [classes.drawerOpen]: open,
            [classes.drawerClose]: !open,
          }) }
          classes={ {
            paper: clsx(classes.paper, {
              [classes.drawerOpen]: open,
              [classes.drawerClose]: !open,
            }),
          } }
        >
          <div className={ classes.toolbar }>
            <IconButton onClick={ handleDrawerClose }>
              { theme.direction === 'rtl' ? <ChevronRightIcon/> : <ChevronLeftIcon/> }
            </IconButton>
          </div>
          <Divider/>
          { renderMenu() }
        </Drawer>
        : <SwipeableDrawer
          onOpen={ () => setMobileDrawerOpen(true) }
          classes={ {
            paper: clsx(classes.paper, {
              [classes.drawerOpen]: mobileDrawerOpen,
              [classes.drawerClose]: !mobileDrawerOpen,
            })
          } }
          className={ classes.drawer }
          anchor={ 'left' }
          open={ mobileDrawerOpen }
          onClose={ () => setMobileDrawerOpen(false) }
        >
          { renderMenu() }
        </SwipeableDrawer>
      }
      <main className={ classes.content }>
        <div className={ classes.toolbar }/>
        { children }
      </main>
    </div>
  );
}
