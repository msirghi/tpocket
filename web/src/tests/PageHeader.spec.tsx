import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { PageHeader } from "../components/layout/PageHeader";
import { findByTestAttr } from "./utils/findByTestAttr";
import toJson from 'enzyme-to-json';

describe('PageHeader Component', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<PageHeader primaryText={ 'Primary text' } secondaryText={ 'Secondary Text' }/>);
  });

  it('should render component properly', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Wrapper has text-center class', () => {
    const wrapper = findByTestAttr(component, 'header-wrapper');
    expect(wrapper.hasClass('text-center')).toBeTruthy();
  });
});
