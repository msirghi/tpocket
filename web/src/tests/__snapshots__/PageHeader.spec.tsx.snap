// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Component should render component properly 1`] = `
<div
  className="text-center"
  data-test="header-wrapper"
>
  <WithStyles(ForwardRef(Typography))
    component="h5"
    data-test="primary-text"
    gutterBottom={true}
    variant="h5"
  >
    Primary text
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    data-test="secondary-text"
    gutterBottom={true}
    variant="subtitle1"
  >
    Secondary Text
  </WithStyles(ForwardRef(Typography))>
</div>
`;
